src/divert.cpp

```diff
@@ -223,7 +223,7 @@ void DivertTask::update_state()
     double trigger_current = _evse->getMinCurrent() * min(1.0, divert_PV_ratio);
 
     // the smoothed current suffices to ensure a sufficient ratio of PV power
-    if (_smoothed_available_current >= trigger_current)
+    if (_smoothed_available_current >= trigger_current + EVSE_DIVERT_HYSTERESIS)
```

Comment:
The `else` needs an occompianying `if` so there is a 'dead' zone between being active and disabled, eg

```C
    // the smoothed current suffices to ensure a sufficient ratio of PV power
    if (_smoothed_available_current >= trigger_current + EVSE_DIVERT_HYSTERESIS)
    {
      EvseProperties props(EvseState::Active);
      props.setChargeCurrent(_charge_rate);
      _evse->claim(EvseClient_OpenEVSE_Divert, EvseManager_Priority_Divert, props);
    }
    else if (_smoothed_available_current <= trigger_current - EVSE_DIVERT_HYSTERESIS)
    {
      if (EvseState::Active == _evse->getState(EvseClient_OpenEVSE_Divert) && 0 == min_charge_time_remaining)
      {
        EvseProperties props(EvseState::Disabled);
        _evse->claim(EvseClient_OpenEVSE_Divert, EvseManager_Priority_Default, props);
      }
    }
```
It is debatable if `EVSE_DIVERT_HYSTERESIS` should be applied to the upper level, lower level or both but the point of the hysteresis is that the oon value and off vale are not the same

Reply:
thanks, I've missed this one. Not sure we need to apply it when ramping down, or if we need to have it configurable yet. We'll see this with some more tests